x-pack/plugins/global_search_providers/public/providers/application.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { from } from 'rxjs';
+import { take, map, takeUntil, mergeMap } from 'rxjs/operators';
+import { ApplicationStart } from 'src/core/public';
+import { GlobalSearchResultProvider } from '../../../global_search/public';
+import { getAppResults } from './get_app_results';
+
+export const createApplicationResultProvider = (
+  applicationPromise: Promise<ApplicationStart>
+): GlobalSearchResultProvider => {
+  return {
+    id: 'application',
+    find: (term, { aborted$, maxResults }) => {
+      return from(applicationPromise).pipe(
+        mergeMap((application) => application.applications$),
```

Comment:
I wonder if we should cache this value so we don't need to recompute on each search. Same with the filtering that is done in `getAppResults` for chromeless apps, etc. I know the perf impact here is probably minimal, but since this may be executed on each keypress, we should go ahead and do the easy wins if doesn't make the code harder to read.